extensions/ql-vscode/src/legacy-query-server/run-queries.ts

```diff
@@ -1,28 +1,164 @@
 import * as tmp from "tmp-promise";
-import { basename, join } from "path";
+import { basename } from "path";
 import { CancellationToken, Uri } from "vscode";
 import { LSPErrorCodes, ResponseError } from "vscode-languageclient";
 
 import * as cli from "../cli";
-import { DatabaseItem } from "../local-databases";
 import {
-  getOnDiskWorkspaceFolders,
-  showAndLogErrorMessage,
-  showAndLogExceptionWithTelemetry,
-  showAndLogWarningMessage,
-  tryGetQueryMetadata,
-  upgradesTmpDir,
-} from "../helpers";
+  DatabaseContentsWithDbScheme,
+  DatabaseItem,
+  DatabaseResolver,
+} from "../local-databases";
+import { showAndLogExceptionWithTelemetry, upgradesTmpDir } from "../helpers";
 import { ProgressCallback } from "../progress";
 import { QueryMetadata } from "../pure/interface-types";
-import { extLogger, Logger, TeeLogger } from "../common";
+import { extLogger, Logger } from "../common";
 import * as messages from "../pure/legacy-messages";
-import { InitialQueryInfo, LocalQueryInfo } from "../query-results";
+import * as newMessages from "../pure/new-messages";
```

Comment:
Instead of importing from newMessages here, can you put the shared messages in `pure/messages-shared`?

Reply:
The only symbols we reference via `newMessages` are the `QueryResultType` status codes. We translate between the legacy ones and the new ones, and expose only the new ones via the `QueryRunner`.